docs/06-appearance.md.lua

```diff
@@ -0,0 +1,222 @@
+#! /usr/bin/lua
+local args = {...}
+
+local gio = require("lgi").Gio
+local gobject = require("lgi").GObject
+local glib = require("lgi").GLib
+
+local name_attr = gio.FILE_ATTRIBUTE_STANDARD_NAME
+
+-- Recursive file scanner
+local function get_all_files(path, ext, ret)
+    ret = ret or {}
+    local enumerator = gio.File.new_for_path(path):enumerate_children(
+        "FILE_ATTRIBUTE_STANDARD_NAME", 0, nil, nil
+    )
+
+    for file in function() return enumerator:next_file() end do
+        local file_name = file:get_attribute_as_string(name_attr)
+        local file_type = file:get_file_type()
+        if file_type == "REGULAR" and file_name:match(ext or "") then
+            table.insert(ret, path..file_name)
+        elseif file_type == "DIRECTORY" then
+            get_all_files(path..file_name.."/", ext, ret)
```

Comment:
I don't think it matters much, but portability says you shouldn't just concatenate strings, but do what `utils.parse_dir` does: `local child_path = enumerator:get_child(file):get_path()`